anchor/anchorRecord.go

```diff
@@ -17,22 +17,32 @@ import (
 	"github.com/FactomProject/factomd/common/primitives"
 )
 
-//AnchorRecord is used to construct anchor chain
+// CurrentAnchorVersion is the current anchor version
+const CurrentAnchorVersion int = 1
+
+// AnchorRecord is used to construct the anchor chain. The Factom Protocol writes an anchor into
+// a parent blockchain (Bitcoin or Ethereum) approximately every 10 minutes. The accumulated entries
+// from the previous 10 minutes are organized into a Directory Block (DBlock). The information
+// from the directory block is stored in the AnchorRecord.
 type AnchorRecord struct {
-	AnchorRecordVer int
-	DBHeight        uint32 `json:",omitempty"` // The only directory block height included in this anchor
-	KeyMR           string `json:",omitempty"` // Merkle root of the only directory block included in this anchor
+	AnchorRecordVer int    // the only supported version seems to be 1
+	DBHeight        uint32 // Factom Directory Block Height - the unique number associated with this DBlock
+	KeyMR           string // key merkle root of the directory block
 
 	DBHeightMax uint32 `json:",omitempty"` // The highest directory block height included in this anchor window
 	DBHeightMin uint32 `json:",omitempty"` // The lowest directory block height included in this anchor window
 	WindowMR    string `json:",omitempty"` // Merkle root of all directory block KeyMRs from DBHeightMin to DBHeightMax
 
-	RecordHeight uint32 // Directory block height we intended to put the AnchorRecord into
-
+	RecordHeight uint32 // This is the future DBlock height a confirmation of anchoring event X is
+	// written to (usually X+1). In principle, the confirmation status is available
+	// in the Bitcoin/Etherium structs, but RecordHeight acts as an internal syncing
+	// mechanism to ensure that all Factom servers are in sync and working on the same
+	// block (ie, not behind)
```

Comment:
Yeah, this was confusing for me. I had a conversation with Sam on 6/24 in developer discussion on discord and this was my interpretation of what I understood. I'm happy to make any changes requested, or just delete my comments on this particular member completely if its wrong. But I think it needs a comment that is correct, because the parameters usefulness isn't readily apparent.

Reply:
The usefulness of the parameter isn't readily apparent because it isn't actually all that useful :) I only kept it in here for backwards compatibility reasons since it was already here for bitcoin anchors. We will likely remove it in the next iteration of anchoring when we revamp both btc and eth systems to be more decentralized. Steven is right, anchoring has nothing to do with consensus at this point in time.

As for the comment, I'd adapt it to be more like the following:
```
// Likely to be deprecated soon. This is the future DBlock height a confirmation of anchoring event X is
// **intended** to be written to (usually X+1). The field is fairly useless since you can't know for sure
// what block an entry will be included in until after the entry is confirmed, and by that point, you can
// get the block height from it's parent Entry Block.
```

Reply:
I replaced my comments with your comment suggestions above